routing/turns_tts_text.cpp

```diff
@@ -62,11 +67,70 @@ string GetTtsText::GetTurnNotification(Notification const & notification) const
   if (notification.m_useThenInsteadOfDistance && notification.m_turnDir == CarDirection::None)
     return string();
 
-  string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
     return string();
 
   string const distStr = GetTextById(GetDistanceTextId(notification));
+
+  if (notification.m_distanceUnits > 0 && !notification.m_nextStreet.empty()) {
+    // We're going to pronounce the street name.
+
+    // We have to use wstring in order to operate on Unicode text
+    // First, let's get rid of unpronounceable symbols.
```

Comment:
Where are they come from? From the street name? Does it make sense to cleanup this street name only, instead of processing the whole TTS string, and to save a bit of battery?

Reply:
Yes, street name, like `[123] City Center > Main Street` and yes we replace only characters on the street name string, the whole TTS is only assembled later with the ttsOut variable.